substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/jdk/SystemPropertiesSupport.java

```diff
@@ -53,6 +53,7 @@
     /** System properties that are taken from the VM hosting the image generator. */
     private static final String[] HOSTED_PROPERTIES = {
                     "java.version",
+                    "java.runtime.version",
```

Comment:
The `java.runtime.version` value you get when asking for the property at image runtime should not be the same as the `java.runtime.version` value of the JVM that runs the image builder.

In the case of printing `--version` of the `native-image`-image it's ok to bake-in the value from the host system. (We want to show which version of GraalVM built the `native-image`-image and not what `java.runtime.version` runs the image.)

A good runtime value for `java.runtime.version` would contain information that identifies the actual VM running the code that executes the system property lookup. Maybe we should use something like

- `Java Version 11.0.9+10-svm-20.3-b04` or
- `Java Version 11.0.9+10-substratevm-20.3-b04`
- `Java Version 11.0.9+10-native-image-20.3-b04`
- `Java Version 11.0.9+10-native-image-vm-20.3-b04`

@christianwimmer do you have a favorite?